File: trunk/jtemplate/benchmarks/gcd.ml
let _=
	let rec gcd x y=
		if x=y then x
		else
			if x<y then
				gcd x (y-x)
			else
				gcd (x-y) x
	in let rec loop=function
		| 0 -> ()
		| n -> let _=gcd 28388383 100101 in loop (n-1)
	in loop 10000

File: trunk/jtemplate/benchmarks/gcd.rb
def gcd(a,b)
	return a<b ? gcd(a,b-a) : a > b ? gcd(a-b,b) : a
end


a=0
for i in 1..10000 do
	a=gcd(28388383,100101)
end

print(a)
